apps/myaccount/src/hooks/use-resend-account-confirmation-alert.tsx

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ProfileConstants } from "@wso2is/core/constants";
+import {  LinkButton } from "@wso2is/react-components";
+import { AxiosError } from "axios";
+import React, { useEffect, useMemo, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { Dispatch } from "redux";
+import { getPreference } from "../api";
+import { resendVerificationLinkOrCode } from "../api/resend-verification";
+import { ProfileConstants as MyAccountProfileConstants, RecoveryScenario } from "../constants";
+import { SCIMConfigs } from "../extensions/configs/scim";
+import { AlertLevels, AuthStateInterface, SystemNotificationAlertState } from "../models";
+import { PreferenceConnectorResponse, PreferenceProperty, PreferenceRequest } from "../models/preference";
+import { AppState } from "../store";
+import { addAlert } from "../store/actions";
+import "../components/shared/system-notification-alert.scss";
+
+/**
+ * Hook to determine if the account-confirmation alert should show,
+ * and to build its contents and behavior.
+ */
+export const useResendAccountConfirmationAlert = (
+    componentId: string = "resend-account-confirmation-alert"
+): SystemNotificationAlertState => {
+    const { t } = useTranslation();
+    const dispatch: Dispatch = useDispatch();
+
+    const profileDetails: AuthStateInterface = useSelector((state: AppState) => state.authenticationInformation);
+
+    const [ isSelfSignUpSendOtpInEmailEnabled, setIsSelfSignUpSendOtpInEmailEnabled ] = useState<boolean>(false);
+    const [ isAccountStatePendingSelfRegistration,
+        setIsAccountStatePendingSelfRegistration ] = useState<boolean>(false);
```

Comment:
Shall we keep this in one line if possible?

Reply:
It exceeds the max length allowed